woocommerce/woocommerce-ios
# Order Creation: Connect status picker

Closes: https://github.com/woocommerce/woocommerce-ios/issues/5406.

## Description

This PR connects navigation from NewOrder view to StatusList so user can pick different status.

## Changes

- Adds SwiftUI wrapper for `OrderStatusListViewController`.
- Replaces separate `var orderStatus` with observation of `orderDetails` to keep `StatusBadgeViewModel` in sync with model.
- Adds navigation to `OrderStatusList` and its callback handling to update `orderDetails.status`.
- Adds project folders for StatusSection and FlowCoordinator under "Order Creation".

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. (Tap "Create order" in the bottom sheet if that appears.)
4. Verify that status section appears with "Pending payment" status.
5. Tap "Edit" button to display status picker.
6. Select different status and tap "Apply".
7. Verify that status section updates to display selected status.

### Considerations

This change will enable "Create" button, but its navigation and saving any of order data is not working now.

## Screenshots

statuses list|updated status
--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/144595061-b291836f-dc67-4412-bf13-9adf9eb2a714.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/144595066-67a82917-9d19-44bb-888c-59555a03d287.png)

---
